nodejs/eks/cni.ts

```diff
@@ -114,120 +106,23 @@ export interface VpcCniOptions {
-    const tmpKubeconfig = tmp.fileSync();
-    fs.writeFileSync(tmpKubeconfig.fd, args.kubeconfig);
-
-    // Compute the required CNI YAML and dump it to a file.
-    const tmpYaml = tmp.fileSync();
-    fs.writeFileSync(tmpYaml.fd, computeVpcCniYaml(cniYamlText, args));
-
-    // Call kubectl to apply the YAML.
-    childProcess.execSync(`kubectl apply -f ${tmpYaml.name}`, {
-        env: { ...process.env, "KUBECONFIG": tmpKubeconfig.name },
-    });
-}
+// Dynamic providers don't currently work well with multi-language components [1]. To workaround this, the
+// dynamic providers in this library have been ported to be custom resources implemented by the new provider
+// plugin. An alias is used and care has been taken in the implementation inside the provider to avoid any
+// diffs for any existing stacks that were created using the old dynamic provider.
+// [1] https://github.com/pulumi/pulumi/issues/5455
 
 /**
  * VpcCni manages the configuration of the Amazon VPC CNI plugin for Kubernetes by applying its YAML chart. Once Pulumi is
  * able to programatically manage existing infrastructure, we can replace this with a real k8s resource.
  */
-export class VpcCni extends pulumi.dynamic.Resource {
+export class VpcCni extends pulumi.CustomResource {
     constructor(name: string, kubeconfig: pulumi.Input<any>, args?: VpcCniOptions, opts?: pulumi.CustomResourceOptions) {
-        // Check to ensure that kubectl is installed, as we'll need it in order to deploy k8s resources below.
-        try {
-            which.sync("kubectl");
-        } catch (err) {
-            throw new Error("Could not set VPC CNI options: kubectl is missing. See https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl for installation instructions.");
-        }
-
-        const yamlPath = path.join(__dirname, "cni", "aws-k8s-cni.yaml");
-        const cniYamlText = fs.readFileSync(yamlPath).toString();
-
-        const provider = {
-            check: (state: any, inputs: any) => Promise.resolve({inputs: inputs, failedChecks: []}),
-            diff: (id: pulumi.ID, state: any, inputs: any) => Promise.resolve({}),
-            create: (inputs: any) => {
-                applyVpcCniYaml(cniYamlText, <VpcCniInputs>inputs);
-                return Promise.resolve({id: crypto.randomBytes(8).toString("hex"), outs: {}});
-            },
-            update: (id: pulumi.ID, state: any, inputs: any) => {
-                applyVpcCniYaml(cniYamlText, <VpcCniInputs>inputs);
-                return Promise.resolve({outs: {}});
-            },
-            read: (id: pulumi.ID, state: any) => Promise.resolve({id: id, props: state}),
-            delete: (id: pulumi.ID, state: any) => Promise.resolve(),
-        };
-
+        // This was previously implemented as a dynamic provider, so alias the old type.
+        const aliasOpts = { aliases: [{ type: "pulumi-nodejs:dynamic:Resource" }] };
+        opts = opts ? pulumi.mergeOptions(opts, aliasOpts) : aliasOpts;
```

Comment:
Pretty sure `mergeOptions` handles the undefined case, so you can simplify this
```suggestion
        opts = pulumi.mergeOptions(opts, aliasOpts);
```